graphula-core/src/Graphula.hs

```diff
@@ -190,16 +179,16 @@ instance MonadUnliftIO m => MonadUnliftIO (GraphulaIdempotentT m) where
 instance MonadTrans GraphulaIdempotentT where
   lift = GraphulaIdempotentT . lift
 
-instance (Monad m, MonadIO m, MonadGraphulaFrontend m) => MonadGraphulaFrontend (GraphulaIdempotentT m) where
-  type NodeConstraint (GraphulaIdempotentT m) = NodeConstraint m
-  type Node (GraphulaIdempotentT m) = Node m
+  {-
+  TODO figure out how to support tracking keys.
```

Comment:
Actually with everything else monomorphed, I think you can keep the frontend class and recover idempotent graphs

```haskell
class MonadGraphulaFrontend m where
  type NodeConstraint m a :: * -> Constraint
  -- not sure about this constraint might need to have constraints on the Key as well
  insert :: NodeConstraint m a => Maybe (Key a) -> a -> m (Maybe (Entity a))
  remove :: NodeConstraint m a => Key a -> m ()
```